src/Cli/dotnet/SlnFileExtensions.cs

```diff
@@ -222,6 +243,11 @@ private static void AddSolutionFolders(this SlnFile slnFile, SlnProject slnProje
 
                 var pathToGuidMap = slnFile.GetSolutionFolderPaths(nestedProjectsSection.Properties);
 
+                if (slnFile.HasSolutionFolder(nestedProjectsSection.Properties, slnProject))
+                {
```

Comment:
Should we also add some sort of check with an error if the folder has a project with the desired name?

Reply:
I added the error though the logic was a bit more complex that I'd like. Take a look.

Reply:
Awesome, thank you :)